sisl/viz/plot.py

```diff
@@ -1075,16 +1077,12 @@ def setup_hamiltonian(self, **kwargs):
                 # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
             except Exception:
 
-                HSXfile = os.path.join(self.rootDir, self.struct + ".HSX")
+                HSXfile = os.path.join(self.root_dir, self.struct + ".HSX")
```

Comment:
I think you should try and rely on fdf to read the information. In this way you ensure that you *try* all possibilities.
Now you are relying on the HSX file format. However, the Hamiltonian and overlap matrices *may* be found in 2 other file formats. TSHS and nc. Using fdf you get all files for free and sisl handles which file to use.

Reply:
Yeees in fact the default is to try to read from the fdf file. This is just handling an exception if reading from the fdf failed. 

This is because for some reason that I don't remember I had a case where it was unable to read from the fdf, but reading from the HSX file directly worked fine. Maybe there was something wrong with those specific files, I don't know. I've removed this as I agree that it makes no sense, let's see if the error raises again in some situation.

Reply:
ok, great